src/dotnet/CarbonAware.LocationSources.Azure/src/AzureLocationSource.cs

```diff
@@ -77,27 +40,39 @@ private string ReadFromResource(string key)
         return readerMetaData.ReadToEnd();
     }
 
-    protected virtual Dictionary<string, AzureRegion> GetAzureRegions()
+    protected virtual Dictionary<string, NamedGeoposition>? LoadRegionsFromJson()
     {
-        var data = ReadFromResource("azure-regions.json");
-        List<AzureRegion> regionList = JsonConvert.DeserializeObject<List<AzureRegion>>(data);
-        foreach(AzureRegion regionInfo in regionList)
-        {
-            Console.Write("region");
-            Console.WriteLine(regionInfo);
-        }    
-        // Create mapping
-        var azureRegionsMapping = new Dictionary<string, AzureRegion>();
-        foreach (var region in regionList)
-        {
-            azureRegionsMapping[region.Name] = region;
+        var data = ReadFromResource("CarbonAware.LocationSources.Azure.azure-regions.json");
+        List<NamedGeoposition> regionList = JsonSerializer.Deserialize<List<NamedGeoposition>>(data, options) ?? new List<NamedGeoposition>();
+        Dictionary<string, NamedGeoposition> namedGeopositions = new Dictionary<String, NamedGeoposition>();
+        foreach(NamedGeoposition region in regionList) {
+            namedGeopositions.Add(region.RegionName, region);
         }
+        return namedGeopositions;
```

Comment:
Why did we stop caching the results here? Since we instantiate this within a data source, I could see a scenario where we may want to re-generate/update the mapping without having to instantiate a new version of the location source and consequently a new version of the data source

Reply:
We are still caching the results. Lets talk about this in our sync